### ISSUE TITLE
Panic on invalid response

### ISSUE THREAD

**Issue by igorwwwwwwwwwwwwwwwwwwww** [eBay/fabio#159]
**Fabio version:** 1.3.1

**Consul service definition:**

```
{
    "service": {
        "name": "web",
        "tags": ["urlprefix-/"],
        "port": 7778,
        "check": {"script": "true", "interval": "10s"}
    }
}
```

Note that there is nothing listening on that port.

**Curl command:**

```
curl http://localhost:9999
```

**Error:**

```
2016/09/11 18:05:01 http: panic serving [::1]:64335: runtime error: invalid memory address or nil pointer dereference
goroutine 99 [running]:
net/http.(*conn).serve.func1(0xc420070c80)
       	/usr/local/Cellar/go/1.7.1/libexec/src/net/http/server.go:1491 +0x12a
panic(0x3da840, 0xc42000c0f0)
       	/usr/local/Cellar/go/1.7.1/libexec/src/runtime/panic.go:458 +0x243
github.com/eBay/fabio/proxy.(*meteredRoundTripper).RoundTrip(0xc42022c9d0, 0xc420166a50, 0xf, 0xc42022c921, 0x3)
       	/private/tmp/fabio-20160909-52715-13mnwwi/fabio-1.3.1/src/github.com/eBay/fabio/proxy/http.go:28 +0x7f
net/http/httputil.(*ReverseProxy).ServeHTTP(0xc4201a3040, 0x630da0, 0xc4201b69c0, 0xc420166960)
       	/usr/local/Cellar/go/1.7.1/libexec/src/net/http/httputil/reverseproxy.go:214 +0x335
github.com/eBay/fabio/proxy.(*httpProxy).ServeHTTP(0xc4200da0e0, 0x630da0, 0xc4201b69c0, 0xc420166960)
       	/private/tmp/fabio-20160909-52715-13mnwwi/fabio-1.3.1/src/github.com/eBay/fabio/proxy/proxy.go:64 +0x2c2
net/http.serverHandler.ServeHTTP(0xc420107000, 0x630da0, 0xc4201b69c0, 0xc420166960)
       	/usr/local/Cellar/go/1.7.1/libexec/src/net/http/server.go:2202 +0x7d
net/http.(*conn).serve(0xc420070c80, 0x631560, 0xc4201a2fc0)
       	/usr/local/Cellar/go/1.7.1/libexec/src/net/http/server.go:1579 +0x4b7
created by net/http.(*Server).Serve
       	/usr/local/Cellar/go/1.7.1/libexec/src/net/http/server.go:2293 +0x44d
```

I hope that's enough information for you to reproduce the issue.

**Comment by magiconair** [comment]
Could you add the config from the log please and maybe the startup parameters?

**Comment by magiconair** [comment]
I think I know what the problem is. I'll fix it later and push a patch. Thanks for submitting

**Comment by magiconair** [comment]
I thought I had fixed that one already. The `meteredRoundtripper` is not checking whether the `http.Response` is valid before trying to get the `statusCode`.

**Comment by magiconair** [comment]
Merged and released v1.3.2

**Comment by igorwwwwwwwwwwwwwwwwwwww** [comment]
Awesome, thanks for the quick response and fix! 💖
